psycopg2_extension/management/commands/cleandb.py

```diff
@@ -1,17 +1,44 @@
+from psycopg2_extension.conf import settings
 from psycopg2_extension.base_command import PsycopgBaseCommand
 
 
 class Command(PsycopgBaseCommand):
-    help = 'Clean database (vacuum and reindex)'
+    help = 'Clean database (VACUUM and REINDEX)'
+
+    def _run_clean_command(self, connection, cursor, db_query):
+        self.stdout.write(4 * ' ' + 'Running "{}"'.format(db_query))
+        cursor.execute(db_query)
+        for notice in connection.notices:
+            for notice_line in notice.split('\n'):
+                self.stdout.write(8 * ' ' + notice_line)
+
+        # Stdout of PostgreSQL commands must be cleaned to not print the same more times
+        del connection.notices[:]
+
+    def _get_all_tables(self, cursor):
+        cursor.execute(
+            "SELECT table_name FROM information_schema.tables "
+            "WHERE table_schema='public' AND table_type = 'BASE TABLE';"
+        )
+        return [result[0] for result in cursor.fetchall()]
 
     def handle(self, *args, **options):
         db_object = self._get_db_object(**options)
-        conn = self._create_connection(db_object)
-
-        with conn.cursor() as cursor:
-            vacuum_query = 'VACUUM ANALYZE'
-            self.stdout.write('Running "{}"'.format(vacuum_query))
-            cursor.execute(vacuum_query)
-            reindex_query = 'REINDEX DATABASE {}'.format(db_object['NAME'])
-            self.stdout.write('Running "{}"'.format(reindex_query))
-            cursor.execute(reindex_query)
+        connection = self._create_connection(db_object)
+
+        with connection.cursor() as cursor:
+            for table_name in self._get_all_tables(cursor):
```

Comment:
maybe return a set from `_get_all_tables`? then you can use `self._get_all_tables(cursor) - self.EXCLUDE_TABLES` instead of the if

Reply:
good idea

Reply:
But I can keep tables order. So this is not possible